bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/ICalendarHandler.java

```diff
@@ -83,7 +83,7 @@ public ICalendarHandler(Thing thing, HttpClient httpClient, @Nullable EventPubli
         super(thing);
         this.httpClient = httpClient;
         calendarFile = new File(ConfigConstants.getUserDataFolder() + File.separator
-                + getThing().getUID().getAsString().replaceAll("[^a-zA-Z0-9\\._-]", "_") + ".ical");
+                + getThing().getUID().getAsString().replaceAll("[<>:\"/\\\\\\|\\?\\*]", "_") + ".ical");
```

Comment:
The regex here should be reduced to 

````
"[<>:\"/\\\\|?*]"
````

Reply:
OK